src/cmd/linuxkit/pkglib/docker.go

```diff
@@ -166,3 +177,142 @@ func (dr dockerRunner) save(tgt string, refs ...string) error {
 	args := append([]string{"image", "save", "-o", tgt}, refs...)
 	return dr.command(args...)
 }
+
+func (dr dockerRunner) manifestPush(img string, trust bool) error {
+	imgParts := strings.Split(img, ":")
+	if len(imgParts) < 2 {
+		return fmt.Errorf("image not composed of <repo>:<tag> '%s'", img)
+	}
+	repo := imgParts[0]
+	tag := imgParts[1]
+
+	cfgFile := config.LoadDefaultConfigFile(os.Stderr)
+	auth, err := cfgFile.GetAuthConfig(registry)
+	if err != nil {
+		return fmt.Errorf("unable to get auth for %s: %v", registry, err)
+	}
+
+	args := []string{
+		"push",
+		"from-args",
+		"--ignore-missing",
+		"--platforms",
+		strings.Join(platforms, ","),
+		"--template",
+		fmt.Sprintf("%s-ARCH", img),
+		"--target",
+		img,
+	}
+	manTool := "manifest-tool"
+	// we do this separately to avoid printing username and password to debug output
+	log.Debugf("Executing (will add username/password): %v", append([]string{manTool}, args...))
+	args = append([]string{
+		"--username",
+		auth.Username,
+		"--password",
+		auth.Password,
+	}, args...)
+	cmd := exec.Command(manTool, args...)
+
+	var stdout bytes.Buffer
+	cmd.Stdout = &stdout
+	cmd.Stderr = os.Stderr
+	cmd.Env = os.Environ()
+
+	if err := cmd.Run(); err != nil {
+		return fmt.Errorf("failed to execute manifest-tool: %v", err)
+	}
+
+	if !trust {
+		fmt.Printf("trust disabled, not signing %s\n", img)
+		return nil
+	}
+
+	// get the image hash and the length from the manifest tool output
+	manToolOut := string(stdout.Bytes())
+	manToolOutParts := strings.Fields(manToolOut)
+	if len(manToolOutParts) < 3 {
+		return fmt.Errorf("manifest-tool output was less then required 3 parts '%s'", manToolOut)
+	}
+	hashParts := strings.Split(manToolOutParts[1], ":")
+	if len(hashParts) < 2 {
+		return fmt.Errorf("manifest-tool output hash was not in format <repo>:<hash> '%s'", manToolOutParts[1])
+	}
+	hash := hashParts[1]
+	length := manToolOutParts[2]
+
+	notaryAuth := base64.StdEncoding.EncodeToString([]byte(fmt.Sprintf("%s:%s", auth.Username, auth.Password)))
+	// run the notary command to sign
+	args = []string{
+		"-s",
+		notaryServer,
+		"-d",
+		path.Join(os.Getenv("HOME"), ".docker/trust"),
+		"addhash",
+		"-p",
+		fmt.Sprintf("docker.io/%s", repo),
+		tag,
+		length,
+		"--sha256",
+		hash,
+		"-r",
+		"targets/releases",
+	}
+	cmd = exec.Command("notary", args...)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	cmd.Env = append(os.Environ(), fmt.Sprintf("%s=%s", notaryDelegationPassphraseEnvVar, os.Getenv(dctEnvVar)), fmt.Sprintf("%s=%s", notaryAuthEnvVar, notaryAuth))
+	log.Debugf("Executing: %v", cmd.Args)
+
+	if err := cmd.Run(); err != nil {
+		return fmt.Errorf("failed to execute notary-tool: %v", err)
+	}
+
+	// report output
+
+	return nil
+
+}
+
+/*
```

Comment:
I would prefer to not have dead code here.

Reply:
It is all a comment. I went through the effort of aligning it, then realized we had the vendor versioning issues. Is there a better place to keep it?

Reply:
Removed